src/physics/em/EPlusGGInteractor.i.hh

```diff
@@ -24,38 +31,96 @@ EPlusGGInteractor::EPlusGGInteractor(const EPlusGGInteractorPointers& shared,
-    Secondary* secondaries = this->allocate_(0); // XXX
+    // Allocate space for two gammas
+    Secondary* secondaries = this->allocate_(2);
     if (secondaries == nullptr)
     {
-        // Failed to allocate space for a secondary
+        // Failed to allocate space for two secondaries
         return Interaction::from_failure();
     }
 
-    // XXX sample
-    (void)sizeof(rng);
-
-    // Construct interaction for change to primary (incident) particle
+    // Construct interaction for change to the incident positron
     Interaction result;
-    result.action      = Action::scattered;                     // XXX
-    result.energy      = units::MevEnergy{inc_energy_.value()}; // XXX
-    result.direction   = inc_direction_;
-    result.secondaries = {secondaries, 1}; // XXX
-
-    // Save outgoing secondary data
-    secondaries[0].def_id    = shared_.electron_id; // XXX
-    secondaries[0].energy    = units::MevEnergy{0}; // XXX
-    secondaries[0].direction = {0, 0, 0};           // XXX
+    result.action      = Action::absorbed;
+    result.energy      = units::MevEnergy{0}; 
+    result.direction   = {0, 0, 0};
+    result.secondaries = {secondaries, 2};
+
+    // Sample two gammas
+    secondaries[0].def_id = secondaries[1].def_id = shared_.gamma_id;
+
+    const real_type inc_energy = inc_energy_.value();
+    IsotropicDistribution<real_type> gamma_dir;
+
+    if(inc_energy == 0) 
+    {
+        // Save outgoing secondary data
+        secondaries[0].energy = secondaries[1].energy 
+            = units::MevEnergy{shared_.electron_mass};
+
+        secondaries[0].direction = gamma_dir(rng); 
+        for(int i = 0 ; i < 3 ; ++i) {
+            secondaries[1].direction[i] = -secondaries[0].direction[i]; 
+        }
+
+	// Note: polarization is intentionally not implemented
+    }
+    else 
+    {
+        real_type tau     = inc_energy/shared_.electron_mass;
+        real_type gam     = tau + 1.0;
+        real_type tau2    = tau + 2.0;
+        real_type sqgrate = std::sqrt(tau/tau2)*0.5;
+        real_type sqg2m1  = std::sqrt(tau*tau2);
+
+        // Evaluate limits of the energy sampling
+        real_type epsilmin = 0.5 - sqgrate;
+        real_type epsilmax = 0.5 + sqgrate;
+        real_type epsilqot = epsilmax/epsilmin;
+
+        // Sample the energy rate of the created gammas
+        real_type epsil, acceptance_prob;
+        do 
+        {
+	    epsil = epsilmin
+                * std::exp(std::log(epsilqot)*generate_canonical(rng));
+            acceptance_prob = 1. - epsil + (2.*gam*epsil-1.)/(epsil*tau2*tau2);
+        } while (BernoulliDistribution(acceptance_prob)(rng));
+
+        // Scattered Gamma angles
+        real_type cost = (epsil*tau2-1.0)/(epsil*sqg2m1);
+        CHECK(fabs(cost) <= 1.0);
+
+	// Kinematic of the gamma pair
+	real_type total_energy = inc_energy + 2.0*shared_.electron_mass;
+        real_type gamma_energy = epsil*total_energy;
+	real_type eplus_moment = std::sqrt(inc_energy*(total_energy));
```

Comment:
Extra parens around `total_energy` should be removed, and this whole file needs to be run through `clang-format`.

Reply:
Done.  I also ran scripts/dev/install-commit-hooks.sh.